### PR TITLE
fix(#104): fix fs undefined in node < 12 

### PR DESCRIPTION
**Pull request by moogacs**

### CODE REVIEW

File: cli/lib/errors.js [hunk 1]
```diff
@@ -2,5 +2,6 @@
 
 module.exports = {
   CouldNotReadConfigFile: class extends Error {},
-  CouldNotParseConfigToml: class extends Error {}
```
Comment (viktor-ku) [inline cli/lib/errors.js]: I am using this error message in this PR https://github.com/entropic-dev/entropic/pull/186

Comment (varjmes) [reply cli/lib/errors.js]: The error isn't removed. See the code below this line

**Comment by moogacs** [comment]
I brought it back ?

File: cli/lib/config.js [hunk 2]
```diff
@@ -2,35 +2,37 @@
...
-
-  var parsed = null;
-  try {
-    parsed = toml.parse(content);
-  } catch (e) {
-    throw new errors.CouldNotParseConfigToml(filename, e);
-  }
+  var content = '';
+  var parsed = '';
+  await fs.readFile(filename, 'utf8', (e, data) => {
```
Comment (olingern) [inline cli/lib/config.js]: I think using the fs.promises ( `require('fs').promises` ) is cleaner for this. Also, here you're `await`ing a callback.

If you need to do some specific handling for node versions, I would use [`process.version`](https://nodejs.org/api/process.html#process_process_version) and handle accordingly. 

Though, I know the docs say that Node 12 is required, so it's worth asking @chrisdickinson if this is needed.

**Comment by moogacs** [comment]
@chrisdickinson , We need your opinion 😊

File: cli/lib/config.js (hunk 2 above)
Comment (viktor-ku) [inline cli/lib/config.js]: Why would you have a callback and `await` used together?

File: cli/lib/config.js [hunk 3]
```diff
@@ -2,35 +2,37 @@
...
-    parsed = toml.parse(content);
-  } catch (e) {
-    throw new errors.CouldNotParseConfigToml(filename, e);
-  }
+  var content = '';
+  var parsed = '';
+  await fs.readFile(filename, 'utf8', (e, data) => {    
+    if (e.code === 'ENOENT') {      
+      if (!fs.existsSync(filename)){
+        fs.mkdirSync(filename);
```
Comment (viktor-ku) [inline cli/lib/config.js]: Why do you need to make directory?

Comment (moogacs) [reply cli/lib/config.js]: Suppose to be the dir for .toml files 🤔

File: cli/lib/config.js (hunk 2 above)
Comment (moogacs) [reply cli/lib/config.js]: That's true, doesn't make sense , shall i update it or will go with #186

Comment (viktor-ku) [reply cli/lib/config.js]: I'd wait for #186

File: cli/lib/config.js (hunk 3 above)
Comment (viktor-ku) [reply cli/lib/config.js]: I see. Good intention, but as of now I see only one file in the home directory to be created + one in the root of the project.

**Comment by moogacs** [comment]
@viktor-ku  well, after reverting and adding some new stuff , it seems got messy !

**Comment by moogacs** [comment]
@zacanger 👍 that's better

**Comment by moogacs** [comment]
@zacanger that's good 👍

**Comment by moogacs** [comment]
@ceejbot already added with #216   , the only different thing now is #211 , so I don't if shall I revert all the previous commits then push again to resolve those conflicts , 
#104 not decided yet

**Pull request closed without merging**